core/embed/rust/src/ui/model_tt/component/keyboard/passphrase.rs

```diff
@@ -124,25 +174,16 @@ impl Component for PassphraseKeyboard {
         let confirm_btn_area = key_grid.cell(14);
         let back_btn_area = key_grid.cell(12);
 
+        self.key_grid = key_grid;
+
         self.page_swipe.place(bounds);
         self.input.place(input_area);
         self.confirm.place(confirm_btn_area);
         self.back.place(back_btn_area);
         self.scrollbar.place(scroll_area);
         self.scrollbar
             .set_count_and_active_page(PAGE_COUNT, STARTING_PAGE);
-        for (key, btn) in self.keys[self.scrollbar.active_page].iter_mut().enumerate() {
-            // Assign the keys in each page to buttons on a 5x3 grid, starting from the
-            // second row.
-            let area = key_grid.cell(if key < 9 {
-                // The grid has 3 columns, and we skip the first row.
-                key + 3
-            } else {
-                // For the last key (the "0" position) we skip one cell.
-                key + 1 + 3
-            });
-            btn.place(area);
-        }
+        self.place_current_buttons();
```

Comment:
Can you just place all the pages here? IIUC there's no need to redo this after every page change.

Reply:
Thanks for the suggestion, done in [f409b83](https://github.com/trezor/trezor-firmware/pull/2544/commits/f409b8386cbe5fdb4dc4ba93211f802acb213a62)